modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutLookAndFeelDisplayContext.java

```diff
@@ -439,6 +441,21 @@ private JSONObject _getCETJSONObject(
 			() -> typeSettingsUnicodeProperties.getProperty("loadType", null)
 		).put(
 			"name", cet.getName(_themeDisplay.getLocale())
+		).put(
+			"scriptElementAttributesJSON",
```

Comment:
Hey @thiago-buarqque I've been playing around with the JS to try to make is simpler and I think the solution is actually moving the logic from the JS to here instead. 

Rather than pass all the `scriptElementAttributesJSON` to the frontend we can add `predefinedLoadType` and `loadTypeOptions`. We could calculate those values here and then the frontend logic will end up much simpler. What do you think?

Reply:
Also if we were to keep the logic in the JS I refactored it [here](https://github.com/liferay/liferay-portal/commit/a47f973e79869edc6b26ea6c13fa5de817ae9cd2).

Reply:
Hey @ethib137. 

We can't do it on the backend. The file you mentioned is called when the page loads. But when the client extension is chosen from the modal, it doesn't request anything else from the backend. The content that the frontend sees is the one passed from `getPayload` method inside [CETItemDescriptor](https://github.com/liferay-platform-experience/liferay-portal/blob/310be97ac4972bd31896b745524eb152d3c3a8ff/modules/apps/client-extension/client-extension-type-item-selector-web/src/main/java/com/liferay/client/extension/type/item/selector/web/internal/item/selector/CETItemDescriptor.java#L47). So if we were to build that on the BE, we would replicate the same logic on both `CETItemDescriptor` and `LayoutLookAndFeelDisplayContext`.

In summary, the FE looks the way to go. I'll apply the refactor you suggested tomorrow, I had to go home earlier today.

Reply:
Okay... that makes sense. Thanks for looking into it. Be sure to test my changes to make sure I didn't break anything. Thanks.

Reply:
@ethib137 just made the changes, things should be fine now. Thanks for the refactor on the frontend